cmd/gitbase/command/server.go

```diff
@@ -196,53 +200,128 @@ func (c *Server) registerDrivers() error {
 }
 
 func (c *Server) addDirectories() error {
-	if len(c.Git) == 0 && len(c.Siva) == 0 {
-		logrus.Error("At least one git folder or siva folder should be provided.")
+	if len(c.Directories) == 0 {
+		logrus.Error("At least one folder should be provided.")
 	}
 
-	for _, pattern := range c.Git {
-		if err := c.addGitPattern(pattern); err != nil {
-			return err
-		}
+	if c.DisableGit && c.DisableSiva {
+		logrus.Warn("The load of git repositories and siva files are disabled," +
+			" no repository will be added.")
+
+		return nil
+	}
+
+	if c.Depth < 1 {
+		logrus.Warn("--depth flag set to a number less than 1," +
+			" no repository will be added.")
+
+		return nil
 	}
 
-	for _, pattern := range c.Siva {
-		if err := c.addSivaPattern(pattern); err != nil {
+	for _, directory := range c.Directories {
+		if err := c.addDirectory(directory); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
-func (c *Server) addGitPattern(pattern string) error {
-	prefix, matches, err := gitbase.PatternMatches(pattern)
+func (c *Server) addDirectory(directory string) error {
+	matches, err := gitbase.PatternMatches(directory)
 	if err != nil {
 		return err
 	}
 
-	for _, m := range matches {
-		logrus.WithField("dir", m).Debug("git repositories directory added")
-		if err := c.pool.AddDir(prefix, m); err != nil {
-			return err
+	for _, match := range matches {
+		if err := c.addMatch(match); err != nil {
+			logrus.WithFields(logrus.Fields{
+				"path":  match,
+				"error": err,
+			}).Error("path couldn't be inspected")
 		}
 	}
 
 	return nil
 }
 
-func (c *Server) addSivaPattern(pattern string) error {
-	matches, err := filepath.Glob(pattern)
+func (c *Server) addMatch(match string) error {
+	root, err := filepath.Abs(match)
 	if err != nil {
 		return err
 	}
 
-	for _, m := range matches {
-		logrus.WithField("dir", m).Debug("siva repositories directory added")
-		if err := c.pool.AddSivaDir(m); err != nil {
+	initDepth := strings.Count(root, string(os.PathSeparator))
+	return filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
 			return err
 		}
+
+		if info.IsDir() {
```

Comment:
You could split these two cases to their own functions, and maybe then simplify some of the conditions with return early clauses.

Reply:
I've placed the logic related to add a git repository in its own method, but I can see how to simplify the conditions